packages/components/src/dropdown-menu/index.js

```diff
@@ -16,20 +17,47 @@ import IconButton from '../icon-button';
 import Dropdown from '../dropdown';
 import { NavigableMenu } from '../navigable-container';
 
+function mergeProps( props = {}, defaultProps = {} ) {
```

Comment:
`Object.assign`, `lodash.assign` or `jQuery.extend` - they mutate the first param. This helper is pure in that regard. I can swap params if you think that it will make it easier to read. and change the logic in a way where `defaultProps` is used as a base for mutations.

Yes, it feels like we have other places where we can reuse the same pattern.

Reply:
Updated in bb25919abca92e684925cdc7946897c5093fb4c0 to follow the order of params you recommended. I didn't enable mutations, we can revisit it later when we start using the same helper on a larger scale.